drake/solvers/NPCPP/BoundingBoxConstraint.cpp

```diff
@@ -0,0 +1,12 @@
+#include "BoundingBoxConstraint.h"
+
+#include <utility>
+#include <vector>
+
+namespace drake {
+BoundingBoxConstraint::BoundingBoxConstraint(snopt::doublereal lb,
```

Comment:
The constraint classes shouldn't know anything about snopt::doublereal.  They should just use doubles.  Only the solve method which is particular to snopt should use the snopt datatypes.  We will have other solvers without snopt that use these same constraint classes.